dapps/src/lib.rs

```diff
@@ -153,30 +153,44 @@ impl ServerBuilder {
 			fetch: None,
 		}
 	}
+}
 
+impl<T: Fetch> ServerBuilder<T> {
 	/// Set a fetch client to use.
-	pub fn with_fetch(&mut self, fetch: FetchClient) {
-		self.fetch = Some(fetch);
+	pub fn fetch<X: Fetch>(self, fetch: X) -> ServerBuilder<X> {
```

Comment:
It would need type hints when calling `ServerBuilder::new()` in such case and builder would be tied to a single implementation of `Fetch`.
Currently you can just do `ServerBuilder::new()` to have a default implementation and then call `fetch(fetch_instance)` to change the type. We use different `fetch` only for tests.